### DISCUSSION TITLE
Doesn't nuxt-content make more sense?

### DISCUSSION THREAD

**Discussion by comfuture** [nuxt/content#1087] [Ideas]
Doesn't it make sense to have a `<nuxt-content />` component instead `<Content />` like the built-in `<nuxt-page />` component?
Auto-route under `pages` directory => `<nuxt-page />`
Auto-route under `content` directory => `<nuxt-content />`

It is also consistent with the web components naming convention of using hyphens in custom component names.

I recognized that single word components were mostly used in nuxt3? Any reason?

**Comment by atinux** [comment]
In the upcoming Content V2 release, we are going to have multiple components:
- `<ContentDoc>` ->  fetches and renders a single document (with `$route.path` by default)
- `<ContentRenderer>` -> renders a document coming from a query.
- `<ContentList>` -> fetches a list of documents and uses slots to render

**Reply by comfuture** [reply]
Good. I am looking forward to seeing what various attempts are being made.

Just to add a little idea,
How about an invisible component that performs queries and a separate component in charge of rendering?
(I know these discussion has appeared before)

```html
<ContentQuery :path="routePath">
  <template #document="doc">
    <ContentDoc :document="doc" />
  </template>
</ContentQuery>
```
A single tag using the $route.path parameter automatically achieves the benefit of being concise but this one may more consistent.

```html
<ContentQuery path="/blog" :find="[{author: 'nuxt', tags: 'release'}]" :only="['title', 'author']">
  <template #result="docs">
    <ul>
      <li v-for="doc in docs" :key="doc.slug">{{ doc.title }} - by {{ doc.author }}</li>
    </ul>
  </template>
</ContentQuery>
```
